watertap/property_models/unit_specific/activated_sludge/tests/test_modified_asm2d_thermo.py

```diff
@@ -203,24 +262,6 @@ def test_build(self, model):
 
         metadata = model.params.get_metadata().properties
 
-        # check that properties are not built if not demanded
-        for v in metadata.list_supported_properties():
-            if metadata[v.name].method is not None:
-                if model.props[1].is_property_constructed(v.name):
-                    raise PropertyAttributeError(
-                        "Property {v_name} is an on-demand property, but was found "
-                        "on the stateblock without being demanded".format(v_name=v.name)
-                    )
-
-        # check that properties are built if demanded
-        for v in metadata.list_supported_properties():
-            if metadata[v.name].method is not None:
-                if not hasattr(model.props[1], v.name):
-                    raise PropertyAttributeError(
-                        "Property {v_name} is an on-demand property, but was not built "
-                        "when demanded".format(v_name=v.name)
```

Comment:
can you remind me why this has to be removed?

Reply:
Theses tests were originally written to check that the constraints for VSS, ISS, and TSS (which were present before this PR) were only built on-demand, but I've changed them from constraints to expressions, so I don't think these tests don't apply anymore. If we were to keep them, the test that checks that properties are not built if not demanded would fail